airflow/models.py

```diff
@@ -4739,7 +4739,7 @@ def _fail_unfinished_tasks(self, target_state, session=None):
         # a manual state change kicked off through the UI.
         # In that case we want to kill any tasks that are unfinished
         # by setting their current state to the state given by the user
-        if 'dagrun/edit/' in request.url:
+        if 'dagrun/edit/' in getattr(request, 'url', ''):
```

Comment:
url may not be present in request? or is it that request may be None?

Reply:
for safety, maybe 
```suggestion
        if request and 'dagrun/edit/' in getattr(request, 'url', ''):
```

Reply:
```python
>>> from flask import request
>>> dir(request)
[]
>>> type(request)
<class 'werkzeug.local.LocalProxy'>
```

Reply:
```
>>> getattr(request, 'url', '')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kirolosshahat/bcroot/devenv/venv/lib/python2.7/site-packages/Werkzeug-0.16.1-py2.7.egg/werkzeug/local.py", line 348, in __getattr__
    return getattr(self._get_current_object(), name)
  File "/Users/kirolosshahat/bcroot/devenv/venv/lib/python2.7/site-packages/Werkzeug-0.16.1-py2.7.egg/werkzeug/local.py", line 307, in _get_current_object
    return self.__local()
  File "/Users/kirolosshahat/bcroot/devenv/venv/lib/python2.7/site-packages/Flask-0.11.1-py2.7.egg/flask/globals.py", line 37, in _lookup_req_object
    raise RuntimeError(_request_ctx_err_msg)
RuntimeError: Working outside of request context.

This typically means that you attempted to use functionality that needed
an active HTTP request.  Consult the documentation on testing for
information about how to avoid this problem.
```
🤔

Reply:
```python
>>> try:
...     url = request.url
... except:
...     url = ''
...
>>> url
''
```